make devenv with podman: Error response from daemon: bad parameter: link is not supported

### What happened?

Docker desktop is expensive now. I'm trying to get grafana working with podman.

Mostly seems to go well if you have compatibility mode enabled, only the devenv/docker-compose.yaml file contains 'links' sections which have apparently been deprecated in docker for quite a long time, and I'm guessing podman didn't even bother to implement it.

https://docs.docker.com/engine/network/links/

The build error is:

```
 ⠋ Container devenv-telegraf-1                Creating                                                                               0.0s 
Error response from daemon: bad parameter: link is not supported
Error: executing /usr/local/bin/docker-compose up -d --build: exit status 1
make: *** [devenv] Error 1
```


### What did you expect to happen?

docker compose up should succeed

### Did this work before?

no

### How do we reproduce it?

1. install podman
2. run the developer setup instructions in contribute/developer-guide.md
3.


### Is the bug inside a dashboard panel?

no

### Environment (with versions)?

Grafana:   main@HEAD (12.1.0-pre ?)
OS: macOS Sonoma
Browser: N/A


### Grafana platform?

None

### Datasource(s)?

_No response_

--- a/devenv/docker/blocks/auth/authentik/docker-compose.yaml
+++ b/devenv/docker/blocks/auth/authentik/docker-compose.yaml
@@ -35,8 +35,8 @@
       AUTHENTIK_HOST: http://authentik:9000
       AUTHENTIK_INSECURE: "true"
       AUTHENTIK_TOKEN: 77vDKwFWqCAsD9ykbH6vsGuMHtloM6urfeglrR9KRhWFcABtwmGGiQWHQStw
-    links:
-      - "authentik:authentik"
+    depends_on:
+      - "authentik"
 
   authentik:
     image: ${AUTHENTIK_IMAGE:-ghcr.io/goauthentik/server}:${AUTHENTIK_TAG:-2023.10.6}
@@ -61,9 +61,6 @@
       - ./certs:/certs
       - ./custom-templates:/templates
       - ./geoip:/geoip
-    links:
-      - "authentikdb:authentikdb"
-      - "authentikredis:authentikredis"
 
   authentik-worker:
     image: ${AUTHENTIK_IMAGE:-ghcr.io/goauthentik/server}:${AUTHENTIK_TAG:-2023.10.6}
@@ -86,6 +83,3 @@
       - ./certs:/certs
       - ./custom-templates:/templates
       - ./geoip:/geoip
-    links:
-      - "authentikdb:authentikdb"
-      - "authentikredis:authentikredis"

--- a/devenv/docker/blocks/auth/jwt_proxy/docker-compose.yaml
+++ b/devenv/docker/blocks/auth/jwt_proxy/docker-compose.yaml
@@ -25,8 +25,6 @@
       - 8087:8080
     depends_on:
       - oauthkeycloakdb
-    links:
-      - "oauthkeycloakdb:oauthkeycloakdb"
     restart: unless-stopped
 
   oauthproxy:

--- a/devenv/docker/blocks/auth/oauth/docker-compose.yaml
+++ b/devenv/docker/blocks/auth/oauth/docker-compose.yaml
@@ -25,6 +25,4 @@
       - 8087:8080
     depends_on:
       - oauthkeycloakdb
-    links:
-      - "oauthkeycloakdb:oauthkeycloakdb"
     restart: unless-stopped

--- a/devenv/docker/blocks/collectd/docker-compose.yaml
+++ b/devenv/docker/blocks/collectd/docker-compose.yaml
@@ -7,5 +7,5 @@
       GRAPHITE_PREFIX: collectd.
       REPORT_BY_CPU: 'false'
       COLLECT_INTERVAL: 10
-    links:
+    depends_on:
       - graphite

--- a/devenv/docker/blocks/influxdb/docker-compose.yaml
+++ b/devenv/docker/blocks/influxdb/docker-compose.yaml
@@ -17,8 +17,6 @@
 
   telegraf:
     image: telegraf:latest
-    links:
-      - influxdb
     depends_on:
       - influxdb
     volumes:

--- a/devenv/docker/blocks/influxdb1/docker-compose.yaml
+++ b/devenv/docker/blocks/influxdb1/docker-compose.yaml
@@ -14,7 +14,7 @@
 
   fake-influxdb1-data:
     image: grafana/fake-data-gen
-    links:
+    depends_on:
       - influxdb1
     environment:
       FD_SERVER: influxdb1
@@ -23,7 +23,7 @@
 
   telegraf-influxdb1:
     image: telegraf:latest
-    links:
+    depends_on:
       - influxdb1
     volumes:
       - ./docker/blocks/influxdb1/telegraf.conf:/etc/telegraf/telegraf.conf:ro

--- a/devenv/docker/blocks/tempo/docker-compose.yaml
+++ b/devenv/docker/blocks/tempo/docker-compose.yaml
@@ -11,7 +11,7 @@
       JAEGER_SAMPLER_PARAM: 1
     labels:
       namespace: tns
-    logging: 
+    logging:
       driver: loki
       options:
         loki-url: 'http://localhost:3100/api/prom/push'
@@ -27,7 +27,7 @@
     command:
       - '-log.level=debug'
       - 'http://db'
-    links:
+    depends_on:
       - db
     ports:
       - 0.0.0.0:8001:80
@@ -38,7 +38,7 @@
       JAEGER_SAMPLER_PARAM: 1
     labels:
       namespace: tns
-    logging: 
+    logging:
       driver: loki
       options:
         loki-url: 'http://localhost:3100/api/prom/push'
@@ -54,7 +54,7 @@
     command:
       - '-log.level=debug'
       - 'http://app'
-    links:
+    depends_on:
       - app
     ports:
       - 0.0.0.0:8002:80
@@ -65,7 +65,7 @@
       JAEGER_SAMPLER_PARAM: 1
     labels:
       namespace: tns
-    logging: 
+    logging:
       driver: loki
       options:
         loki-url: 'http://localhost:3100/api/prom/push'
@@ -93,7 +93,7 @@
       - --enable-feature=native-histograms
     volumes:
       - ./docker/blocks/tempo/prometheus.yaml:/etc/prometheus.yaml
-    links:
+    depends_on:
       - app
       - db
       - loadgen
@@ -101,7 +101,7 @@
       - "9090:9090"
     labels:
       namespace: monitoring
-    logging: 
+    logging:
       driver: loki
       options:
         loki-url: 'http://localhost:3100/api/prom/push'
@@ -117,7 +117,7 @@
       - "3100:3100"
     labels:
       namespace: monitoring
-    logging: 
+    logging:
       driver: loki
       options:
         loki-url: 'http://localhost:3100/api/prom/push'

--- a/devenv/docker/blocks/traefik/docker-compose.yml
+++ b/devenv/docker/blocks/traefik/docker-compose.yml
@@ -9,7 +9,7 @@ services:
     ports:
       - '80:80'
       - '8080:8080'
-    links:
+    depends_on:
       - grafana-subpath
 
   grafana-subpath:

